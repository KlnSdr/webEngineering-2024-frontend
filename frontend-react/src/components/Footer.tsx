import React from "react";
import Stack from "react-bootstrap/Stack";

interface FooterProps {
  owner: string;
  timestamp: string;
}

const Footer: React.FC<FooterProps> = ({ owner, timestamp }) => {
    return(
        <footer className="text-center bg-body-tertiary pt-3">
            <Stack direction={"horizontal"} gap={5}>
                <text> Erstellt von: {owner} </text>
                <text> Datum: {timestamp} </text>
            </Stack>
        </footer>
    );
}

export default Footer;