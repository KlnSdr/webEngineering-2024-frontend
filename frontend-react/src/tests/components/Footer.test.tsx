import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Footer from "../../components/Footer";

describe("Footer Component", () => {
    test("renders the correct text", () => {
        render(<Footer  owner={"user"} timestamp={"01.01.1999"}/>);

        const footerElement = screen.getByText(/user/);
        expect(footerElement).toBeInTheDocument();
        expect(footerElement).toHaveTextContent("user");

        const timestampElement = screen.getByText(/01.01.1999/);
        expect(timestampElement).toBeInTheDocument();
        expect(timestampElement).toHaveTextContent("01.01.1999");
    });
});