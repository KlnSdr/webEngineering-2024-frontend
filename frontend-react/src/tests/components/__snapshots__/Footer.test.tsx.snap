// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders the correct text 1`] = `
<DocumentFragment>
  <footer
    class="text-center bg-body-tertiary pt-3"
  >
    <div
      class="hstack gap-5"
    >
      <text>
         Erstellt von: user 
      </text>
      <text>
         Datum: 01.01.1999 
      </text>
    </div>
  </footer>
</DocumentFragment>
`;
